src/notification/twitch.ts

```diff
@@ -0,0 +1,54 @@
+import {Link, Store} from '../store/model';
+import {Print, logger} from '../logger';
+import {RefreshableAuthProvider, StaticAuthProvider} from 'twitch-auth';
+import {existsSync, promises, readFileSync} from 'fs';
+import {ChatClient} from 'twitch-chat-client';
+import {config} from '../config';
+
+const twitch = config.notifications.twitch;
+
+let tokenData = {
+	accessToken: twitch.accessToken,
+	expiryTimestamp: 0,
+	refreshToken: twitch.refreshToken
+};
+
+if (existsSync('./twitch.json')) {
+	tokenData = {...JSON.parse(readFileSync('./twitch.json', 'utf-8')), ...tokenData};
+}
+
+const chatClient: ChatClient = new ChatClient(
+	new RefreshableAuthProvider(
+		new StaticAuthProvider(twitch.clientId, tokenData.accessToken),
+		{
+			clientSecret: twitch.clientSecret,
+			expiry: tokenData.expiryTimestamp === null ? null : new Date(tokenData.expiryTimestamp),
+			onRefresh: async ({accessToken, refreshToken, expiryDate}) => {
+				return promises.writeFile('./twitch.json', JSON.stringify({
+					accessToken,
+					expiryTimestamp: expiryDate === null ? null : expiryDate.getTime(),
+					refreshToken
+				}, null, 4), 'utf-8');
+			},
+			refreshToken: tokenData.refreshToken
+		}
+	),
+	{
+		channels: [twitch.channel]
+	}
+);
```

Comment:
I believe this will cause a problem if the user does no have any Twitch auth setup. Perhaps wrap this around a check.